common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -122,6 +131,37 @@ protected void append(LoggingEvent event) {
             System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
             isTruststore = true;
         }
+        if (dataPublisher == null) {
+            Pattern pattern = Pattern.compile("(tcp):\\/\\/([a-zA-Z0-9]+):([0-9]+)");
+            Matcher matcher = pattern.matcher(url);
+            if (matcher.find() && authURLSet == null) {
+                authURLSet = "ssl://" + matcher.group(2).toString() + ":" + (
+                        Integer.parseInt(matcher.group(3).toString()) + 100);
```

Comment:
Hi Dilini,

This was removed by the commits in https://github.com/wso2/shared-analytics/pull/21/files#diff-cf2e351d348612f2af26a4cb69591f32L139, since the data-publisher can initialize using the receiverURLSet.

Thanks,
SajithD